viite-backend/viite-main/src/main/scala/fi/liikennevirasto/viite/util/JunctionImporter.scala

```diff
@@ -22,141 +18,167 @@ class JunctionImporter(conversionDatabase: DatabaseDef) {
+        createJunctionPointParams(point, junction.id, roadwayPointId)
+      }
     }
 
-    junctionPs.executeBatch()
-    junctionPointPs.executeBatch()
-    junctionPs.close()
-    junctionPointPs.close()
+    // Execute batch updates to main database
+    batchUpdateJunctions(junctionParams)
+    batchUpdateJunctionPoints(junctionPointParams)
+
   }
 
   protected def fetchJunctionsFromConversionTable(): Seq[ConversionJunction] = {
-    conversionDatabase.withDynSession {
-      sql"""SELECT L.ID, LIITTYMANRO, solmunro, TO_CHAR(L.VOIMASSAOLOAIKA_ALKU, 'YYYY-MM-DD hh:mm:ss'), TO_CHAR(L.VOIMASSAOLOAIKA_LOPPU, 'YYYY-MM-DD hh:mm:ss'),
-           TO_CHAR(L.MUUTOSPVM, 'YYYY-MM-DD hh:mm:ss'), L.KAYTTAJA, TO_CHAR(L.REKISTEROINTIPVM, 'YYYY-MM-DD hh:mm:ss')
-           FROM LIITTYMA L JOIN SOLMU S ON (ID_SOLMU = S.id)  """
-        .as[ConversionJunction].list
+    runWithConversionDbReadOnlySession {
+      val query = sql"""
+            SELECT l.id, liittymanro, solmunro, TO_CHAR(L.voimassaoloaika_alku, 'YYYY-MM-DD hh:mm:ss'),
+                TO_CHAR(l.voimassaoloaika_loppu, 'YYYY-MM-DD hh:mm:ss'),
+                TO_CHAR(l.muutospvm, 'YYYY-MM-DD hh:mm:ss'), L.kayttaja, TO_CHAR(l.rekisterointipvm, 'YYYY-MM-DD hh:mm:ss')
+            FROM liittyma L JOIN solmu S ON (id_solmu = S.id)
+            """
+        runSelectQuery(query.map(ConversionJunctionScalike.apply))
     }
   }
 
   protected def fetchJunctionPointsFromConversionTable(): Seq[ConversionJunctionPoint] = {
-    conversionDatabase.withDynSession {
-      sql"""SELECT JP.ID, JP.EJ, AP.ID_AJORATA, AP.ETAISYYS, JP.ID_LIITTYMA, TO_CHAR(JP.MUUTOSPVM, 'YYYY-MM-DD hh:mm:ss'), JP.KAYTTAJA, TO_CHAR(JP.REKISTEROINTIPVM, 'YYYY-MM-DD hh:mm:ss')
-           FROM LIITTYMAKOHTA JP
-           JOIN AJORADAN_PISTE AP ON (JP.ID_AJORADAN_PISTE = AP.ID)
-           JOIN LIITTYMA J ON (JP.ID_LIITTYMA = J.ID)
-           WHERE JP.VOIMASSAOLOAIKA_LOPPU IS NULL OR J.VOIMASSAOLOAIKA_LOPPU IS NOT NULL
+    runWithConversionDbReadOnlySession {
+      val query = sql"""
+           SELECT jp.id, jp.ej, ap.id_ajorata, ap.etaisyys, jp.id_liittyma,
+              TO_CHAR(jp.muutospvm, 'YYYY-MM-DD hh:mm:ss'),
+              jp.kayttaja, TO_CHAR(jp.rekisterointipvm, 'YYYY-MM-DD hh:mm:ss')
+           FROM liittymakohta jp
+           JOIN ajoradan_piste ap ON (jp.id_ajoradan_piste = ap.id)
+           JOIN liittyma j ON (jp.id_liittyma = j.id)
+           WHERE jp.voimassaoloaika_loppu IS NULL OR j.voimassaoloaika_loppu IS NOT NULL
       """
-        .as[ConversionJunctionPoint].list
+       runSelectQuery(query.map(ConversionJunctionPointScalike.apply))
     }
   }
 
+  // Create junction parameters for batch update
+  private def createJunctionParams(conversionJunction: ConversionJunction): Seq[Any] = {
+    Seq(
+      conversionJunction.id,
+      conversionJunction.junctionNumber,
+      conversionJunction.nodeNumber,
+      conversionJunction.startDate,
```

Comment:
vanhassa on TimestampOption, tarvittaisko tässä tuo .orNull?
Vai voiko jättää tälle pois vaan?

Reply:
katson tosin tuota vanhaa getConversionJunction-funktiota, mäppäytyvätkö nämä tarkenteet tuon funktion kanssa ylipäänsä?
(jja jojs, niin sit myös kysymykset
"entäs tolle validFrom-kentälle"
ja
"mihin createdTime on tippunut"
)

Reply:
juteltiin: createdTime tulee triggerillä
jja itseasiassa niin päin, että .orNull on turha.